client/src/api/managment-service.ts

```diff
@@ -383,6 +391,28 @@ class ManagmentServiceApi {
     return reports;
   }
 
+  public async getReportsScheduled(): Promise<ReportSummary[]> {
+    await this.refreshTokenIfExpired();
+    const response = await this.axiosInstance.get('/api/v1/reports/scheduled');
+    const reports: ReportSummary[] = response.data;
+    reports.forEach((report) => {
+      if (!VALID_URGENCY_VALUES.includes(report.urgency)) {
+        throw new Error(
+            `Invalid urgency value "${report.urgency}" for report ID 
+            ${report.id}. Allowed values are: ${VALID_URGENCY_VALUES.join(', ')}`,
+        );
+      }
+    });
+
+    return reports;
+  }
+
+  public async getGeneratingReports(): Promise<GeneratingReport[]> {
+    await this.refreshTokenIfExpired();
+    const response = await this.axiosInstance.get('/api/v1/reports/generating');
```

Comment:
Also, in project we use terms of AwaitingGeneration or PendingGeneration, to avoid using verbs in paths